hbase-server/src/test/java/org/apache/hadoop/hbase/mob/TestMobCompactionWithDefaults.java

```diff
@@ -180,6 +180,42 @@ public void baseTestMobFileCompaction() throws InterruptedException, IOException
     LOG.info("MOB compaction " + description() + " finished OK");
   }
 
+  @Test
+  public void testMobFileCompactionAfterSnapshotClone() throws InterruptedException, IOException {
+    final TableName clone = TableName.valueOf(test.getMethodName() + "-clone");
+    LOG.info("MOB compaction of cloned snapshot, " + description() + " started");
+    loadAndFlushThreeTimes(rows, table, famStr);
+    LOG.debug("Taking snapshot and cloning table {}", table);
+    admin.snapshot(test.getMethodName(), table);
+    admin.cloneSnapshot(test.getMethodName(), clone);
```

Comment:
Wait and check if snapshot is done?

Reply:
The `Admin.snpashot` and `Admin.cloneSnasphot` calls are both synchronous. they're only supposed to return if their operations are successful. The should throw an exception if something goes wrong.